lib/shared.js

```diff
@@ -16,13 +16,21 @@ module.exports = {
     // Append 'ts' extensions to Airbnb 'import/resolver' setting
     'import/resolver': {
       node: {
-        extensions: ['.mjs', '.js', '.ts', '.json'],
+        extensions: ['.mjs', '.js', '.json', '.ts', '.d.ts'],
       },
     },
     // Append 'ts' extensions to Airbnb 'import/extensions' setting
-    'import/extensions': ['.js', '.ts', '.mjs'],
+    'import/extensions': ['.js', '.mjs', '.jsx', '.ts', '.tsx', '.d.ts'],
+    // Resolve type definition packages
+    'import/external-module-folders': ['node_modules', 'node_modules/@types'],
+    // Don't ignore 'node_modules' for improved safety
+    'import/ignore': ['\\.(coffee|scss|css|less|hbs|svg|json)$'],
```

Comment:
What does "improved safety" mean? Not sure about this one, because:

1) It doesn't exist in the [plugin:import/typescript rules](https://github.com/benmosher/eslint-plugin-import/blob/master/config/typescript.js)

2) I feel that Airbnb's default config included `node_modules` for a good reason

But I don't fully understand what this change achieves, so please let me know!

Reply:
You're right, the `node_modules` directory should be kept ignored. I thought including it would improve type safety, but that should be left up for TypeScript.